near/src/config.rs

```diff
@@ -610,26 +608,56 @@ impl GenesisConfig {
     }
 
     /// Reads GenesisConfig from a file.
-    pub fn from_file(path: &PathBuf) -> Self {
+    pub fn from_file(path: &PathBuf, records_path: Option<PathBuf>) -> Self {
         let mut file = File::open(path).expect("Could not open genesis config file.");
         let mut content = String::new();
         file.read_to_string(&mut content).expect("Could not read from genesis config file.");
-        GenesisConfig::from(content.as_str())
+        let records_str = records_path.map(|path| {
+            let mut content = String::new();
+            let mut file = File::open(&path).expect("Could not open genesis records file.");
+            file.read_to_string(&mut content).expect("Could not read from genesis records file.");
+            content
+        });
+        GenesisConfig::from(content.as_str(), records_str)
     }
 
     /// Writes GenesisConfig to the file.
     pub fn write_to_file(&self, path: &Path) {
         let mut file = File::create(path).expect("Failed to create / write a genesis config file.");
-        let str =
-            serde_json::to_string_pretty(self).expect("Error serializing the genesis config.");
-        if let Err(err) = file.write_all(str.as_bytes()) {
+        let mut buf =
+            serde_json::to_vec_pretty(self).expect("Error serializing the genesis config.");
+        if let Err(err) = file.write_all(&mut buf) {
+            panic!("Failed to write a genesis config file {}", err);
+        }
+    }
+
+    /// Writes GenesisConfig to the 2 files.
+    /// All fields but records into `config_path` file and records into a `records_path` file.
+    /// Requires `&mut self` to avoid cloning records.
+    pub fn write_to_config_and_records_files(&mut self, config_path: &Path, records_path: &Path) {
+        // Writing records
+        let mut file =
+            File::create(records_path).expect("Failed to create / write a genesis records file.");
+        let mut buf = serde_json::to_vec_pretty(&self.records)
+            .expect("Error serializing the genesis records.");
+        if let Err(err) = file.write_all(&mut buf) {
+            panic!("Failed to write a genesis records file {}", err);
+        }
```

Comment:
Heads up! There is a convenient method in Rust standard library: https://doc.rust-lang.org/stable/std/fs/fn.write.html

```rust
std::fs::write("foo.txt", b"Lorem ipsum")?;
```